lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -453,30 +462,62 @@ async function validateQuotas(bucket, account, apiNames, apiMethod, inflight, lo
             }
         }
 
+        if (actionWithDataDeletion[apiMethod]) {
+            const responseTimeInNs = Number(process.hrtime.bigint() - requestStartTime);
+            monitoring.quotaEvaluationDuration.labels({
+                action: apiMethod,
+                type: 'delete',
+                code: 200,
+            }).observe(responseTimeInNs / 1e9);
+            return callback();
+        }
+
         if (bucketQuotaExceeded || accountQuotaExceeded) {
-            if (apiMethod?.endsWith('Delete')) {
-                return callback();
-            }
             // clean any inflight bytes
             if (inflight > 0) {
                 // eslint-disable-next-line no-await-in-loop
-                await ScubaClientInstance.getLatestMetrics('bucket',
+                await ScubaClientInstance.getUtilizationMetrics('bucket',
                     `${bucket.getName()}_${creationDate}`, null, {
                     action: apiMethod,
                     inflight: -inflight,
                 });
                 if (account?.quota) {
                     // eslint-disable-next-line no-await-in-loop
-                    await ScubaClientInstance.getLatestMetrics('account',
+                    await ScubaClientInstance.getUtilizationMetrics('account',
```

Comment:
would it be more interesting to have another endpoint in scuba to retrieve both bucket and account metrics so you send only one request ?

Reply:
and also where you could pass both `inflight` and `-inflight` for delete to reduce the number of api calls

Reply:
> would it be more interesting to have another endpoint in scuba to retrieve both bucket and account metrics so you send only one request ?

Yes, this is something we can do; for now the use case will be to have only bucket quotas, but in case they have both, we can add another API

But this breaks compatibility with the existing scuba APIs, so better doing that in a future iteration to improve the performance, what do you think?

Reply:
yep, maybe make a ticket so we don't forget

Reply:
I created an epic and the ticket is inside: https://scality.atlassian.net/browse/ARTESCA-12124
We already identified a bunch of things we need to improve, so this epic will be useful for that.